status-core/src/main/java/com/indeed/status/core/AbstractDependencyManager.java

```diff
@@ -327,6 +336,14 @@ public void addDependency(final Dependency dependency) {
     }
 
     public Dependency removeDependency(final String id) {
+        if (dependencyPingers.containsKey(id)) {
```

Comment:
Nit: I don't think you need this first level of indirection; `dependencyPingers.remove(id)` returns null if the key isn't in there.

And removing it is technically more correct because there could be some kind of weird race between checking to see if the map contains the key and actually removing it.

Reply:
Agreed on this. Just remove and check if pinger is null.